pkg/dogstatsd/server_worker.go

```diff
@@ -48,6 +56,10 @@ func (w *worker) run() {
 			// we return the samples in case the slice was extended
 			// when parsing the packets
 			w.samples = w.server.parsePackets(w.batcher, w.parser, packets, w.samples)
+		case notify := <-w.flushChan:
+			w.batcher.flush()
```

Comment:
The call to `flush()` being block, this code change means that the worker won't process any metric while the aggregator is flushing, which is a major caveat and something we can't afford in the main DogStatsD version.

Wasn't  the call to `InitAggregatorWithFlushInterval` with a `0` interval enough?

Reply:
Unfortunately not. `w.server.parsePackets()` also calls `w.batcher.flush()` once it's finished processing. So that meant that `w.batcher.flush` could be called from this 'run' goroutine, and whatever goroutine is calling flush, which was also causing a race condition. `w.batcher.flush` wasn't written to be thread safe. As far as I can tell, the 'flush' method here is only called through serverless code paths.